doc/tutorials/acrn_on_qemu.rst

```diff
@@ -0,0 +1,288 @@
+
+4. Build the Service VM kernel so as to include the virtio-blk driver. User VM (L2 guest) uses virtio-blk
+   driver to mount rootfs.
+
+   a. Please clone the latest stable version or master as show below,
+
+      .. code-block:: none
+
+         $ git clone https://github.com/projectacrn/acrn-kernel
+         $ cd acrn-kernel
+
+   b. Copy kernel_config_uefi_sos to .config
+   c. make olddefconfig
+   d. make menuconfig
+
+      .. figure:: images/acrn_qemu_4.png
+         :align: center
+
+   f. make
+   g. Copy ``arch/x86/boot/bzImage`` to the Service VM /boot/ directory and rename it to ``bzImage_sos``.
+
+   .. note::
+      Since the needed drivers are part of the Service VM kernel itself, we can avoid installing the modules.
+
+5. Update Ubuntu GRUB to boot ACRN hypervisor and load ACRN Kernel Image. Append the following
+   configuration to the :file:`/etc/grub.d/40_custom`,
+
+   .. code-block:: none
+
+      menuentry 'ACRN hypervisor' --class ubuntu --class gnu-linux --class gnu --class os $menuentry_id_option 'gnulinux-simple-e23c76ae-b06d-4a6e-ad42-46b8eedfd7d3' {
+         recordfail
+         load_video
+         gfxmode $linux_gfx_mode
+         insmod gzio
+         insmod part_msdos
+         insmod ext2
+
+         echo 'Loading ACRN hypervisor with SDC scenario ...'
+         multiboot --quirk-modules-after-kernel /boot/acrn.32.out
+         module /boot/bzImage_sos Linux_bzImage
+      }
+
+6. Update GRUB ``sudo update-grub``.
+
+7. Shutdown the guest and relaunch using, ``virsh start ACRNSOS --console`` and select ACRN hypervisor from GRUB menu to launch Service VM running on top of ACRN.
+   This can be verified from ``dmesg`` as shown below,
+
+   .. code-block:: none
+
+      guestl1@ACRNSOS:~$ dmesg | grep ACRN
+      [    0.000000] Hypervisor detected: ACRN
+      [    2.337176] ACRNTrace: Initialized acrn trace module with 4 cpu
+      [    2.368358] ACRN HVLog: Initialized hvlog module with 4 cpu
+      [    2.727905] systemd[1]: Set hostname to <ACRNSOS>.
+
+8. When shutting down, make sure to cleanly destroy the Service VM to prevent crashes in subsequent boots. This can be done using,
+
+      .. code-block:: none
+
+         virsh destory ACRNSOS # where ACRNSOS is the virsh domain name.
+
+
+Service VM Networking updates for User VM
+*****************************************
+If you are planning to enable networking for User VM(L2 guest), the following steps might be helpful.
+
+1. Edit your :file:`/etc/netplan/01-netcfg.yaml` file to add acrn-br0 as below,
+
+   .. code-block:: none
+
+      network:
+         version: 2
+         renderer: networkd
+         ethernets:
+            enp1s0:
+               dhcp4: no
+         bridges:
+            acrn-br0:
+               interfaces: [enp1s0]
+               dhcp4: true
+               dhcp6: no
+
+2. Apply the new network configuration by,
+
+      .. code-block:: none
```

Comment:
ReST is sensitive to extra spaces, so just indent the code block directive by three spaces (not six).  There are some other cases in this doc with excess indentation too.